### PR TITLE
Bring upstream changes for tensorflow

### PR DESCRIPTION
**Pull request by pramodk**
@matz-e : this is going to re-build some of the packages but I believe it will be straightforward.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-scipy/package.py [hunk 1]
```diff
@@ -30,30 +30,35 @@ class PyScipy(PythonPackage):
...
-    depends_on('python@2.6:2.8,3.2:')
-    depends_on('python@2.7:2.8,3.4:', when='@0.18:')
-    depends_on('python@3.5:', when='@1.3:')
+    depends_on('python@2.6:2.8,3.2:', type=('build', 'run'))
+    depends_on('python@2.7:2.8,3.4:', when='@0.18:', type=('build', 'run'))
+    depends_on('python@3.5:', when='@1.3:', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
-    depends_on('py-pytest', type='test')
+    depends_on('py-pybind11@2.4.0:', when='@1.4.1:')
+    depends_on('py-pybind11@2.2.4:', when='@1.4.0:')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-scipy/package.py:55]: @matz-e : FYI, in the upstream we had:

```
-    depends_on('py-pybind11@2.4.0:', when='@1.4.1:', type='build')
-    depends_on('py-pybind11@2.2.4:', when='@1.4.0:', type='build')
```

Note that people often put `build` dependency assuming package is used during building only. This is incorrect and it should be also `link` dependency. IIRC, such things I saw in some bbp packages as well.

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-scipy/package.py:55]: This is a Python package though, which should not really be linked further?

**Review by matz-e — COMMENTED** [review]
I've gone through some files, but honestly, I think we should develop this on top of `merge/2019/10` directly. Pull requests will always fail as they are setup to use our current deployment as upstream.

File: var/spack/repos/builtin/packages/py-scipy/package.py (hunk 1 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/py-scipy/package.py:55]: I did saw compilation/link error and that's when noticed this dependency. (don't want to guess what tensorflow does with it's dependencies :) )

**Comment by pramodk** [comment]
> I've gone through some files, but honestly, I think we should develop this on top of merge/2019/10 directly.

yeah agree. I am creating here draft PR just for discussion/ future reference.

File: var/spack/repos/builtin/packages/py-scipy/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-scipy/package.py:55]: …sigh…

**Comment by matz-e** [comment]
@pramodk still relevant?

**Comment by matz-e** [comment]
@pramodk we're now past another merge with upstream, do we still need this?

**Comment by matz-e** [comment]
Closing this, don't think we need it any longer.

**Pull request closed without merging**